### ISSUE TITLE
Usage of longest-chain instead of most-work

### ISSUE THREAD

**Issue by gregdhill** [interlay/interbtc#1186]
**Describe the bug**
The BTCRelay *incorrectly* sets the main chain as the longest fork instead of that which has the most work. This means that an attacker with significant mining power could mine a chain with an invalid timestamp to lower the difficulty after each readjustment period. It is currently not economically feasible to exploit due to the required mining requirements.

**Expected behavior**
We should only reorg when the fork has more work than the current main chain, see how electrum does that [here](https://github.com/spesmilo/electrum/blob/cee22abcb5544c5a6fa7f8a8108ccda9c32c2e29/electrum/blockchain.py#L582-L614).

**Additional context**
The summa relay [here](https://github.com/summa-tx/relays/blob/30704ab2c443a8a24bdb1228da1e453347f62d3b/solidity/contracts/Relay.sol#L458-L463) only checks the chain work when reorging but it does not support reorgs over 2017 blocks.

**Comment by nakul1010** [comment]
@gregdhill Thanks for the references few questions

- I am expected to extend [this if condition](https://github1s.com/interlay/interbtc/blob/master/crates/btc-relay/src/lib.rs#L1200-L1201) 
    - `&&( get_chainwork(current_fork.max_height) => get_chainwork(prev_chain.max_height))`
- Regarding the [target](https://github1s.com/spesmilo/electrum/blob/cee22abcb5544c5a6fa7f8a8108ccda9c32c2e29/electrum/blockchain.py#L585-L586) var in  electrum is same as in bitcoin types [here](https://github1s.com/interlay/interbtc/blob/master/crates/bitcoin/src/types.rs#L188-L189) ?

**Comment by gregdhill** [comment]
Yes we need to extend that condition and we should also update `ensure_no_ongoing_fork`. I can't see which `target` type you are linking to in electrum, but I am assuming it is the same.
